synapse/storage/databases/main/e2e_room_keys.py

```diff
@@ -38,6 +38,15 @@ class RoomKey(TypedDict):
 
 
 class EndToEndRoomKeyStore(SQLBaseStore):
+    """The store for end to end room key backups.
+
+    See https://spec.matrix.org/v1.1/client-server-api/#server-side-key-backups
+
+    As per the spec, backups are identified by an opaque version string. Internally,
+    version identifiers are assigned using incrementing integers. Non-numeric version
+    strings are treated as if they do not exist, since we would have never issued them.
```

Comment:
I think I said this on the changes in the previous commit, but worried this might let some client requests fail silently while otherwise indicating success.